### PR TITLE
Support contextvars on python>=3.7

### PR DESCRIPTION
**Pull request by zetahernandez**
prevent correlation_id state from bleeding to other code unexpectedly, when used in concurrent code.

### CODE REVIEW

**Comment by dbaty** [comment]
Hi @zetahernandez : thanks for your contribution. This is just a quick acknowledgment that your pull request has been noticed. :) However, some of my fellow maintainers are in vacation, so it may take a few weeks before we can review and merge. ETA: before mid-september.

**Comment by adamantike** [comment]
@dbaty is there any updated ETA on this one?

File: tests/test_context_processors.py [hunk 1]
```diff
@@ -10,12 +10,12 @@ class TestContextProcessor(TestCase):
     @mock.patch('cid.context_processors.get_cid')
     def test_cid_added(self, get_cid):
         get_cid.return_value = 'a-text-cid'
-        self.assertEquals(
+        self.assertEqual(
```
Comment (dbaty) [inline tests/test_context_processors.py:13]: Good catch, thank you!

**Review by dbaty — COMMENTED** [review]
Overall, it looks pretty good. Thanks for your contribution and your patience.

Apart from my comments, a new entry in the changelog is warranted. Something along the lines of:

> Under Python 3.7 and later, use context variables (with the `contextvars` module) instead of a thread-local variable to avoid state bleeding in concurrent code.

**Comment by dbaty** [comment]
@zetahernandez : if you could remove the pinning of isort and add a changelog entry, I would gladly merge this pull request.

File: requirements/quality.txt [hunk 2]
```diff
@@ -1,4 +1,4 @@
 # `readme_renderer` is needed to check for `setup.py check --restructuredtext`
-isort
+isort~=4.3
```
[the remark this answers is not in the document]
Comment (zetahernandez) [reply requirements/quality.txt]: I've experienced some issues with the 5 version of isort when running the ci, but looks to be resolved now.

**Comment by zetahernandez** [comment]
Thanks @dbaty for reviewing! I applied the your comments.

**Comment by dbaty** [comment]
Thanks again for your contribution. I apologize for the delay. For your information, I plan to cut a new release before the end of January. This year, hopefully! ;)

**Comment by adamantike** [comment]
@dbaty, are we getting a new release any time soon? :)

**Comment by dbaty** [comment]
django-cid 2.2. has just been released. :)

**Comment by adamantike** [comment]
Thanks a lot, @dbaty!

**Pull request merged**